lib/phoenix_token_auth/registration_flow.ex

```diff
@@ -0,0 +1,28 @@
+defmodule PhoenixTokenAuth.RegistrationFlow do
+  alias PhoenixTokenAuth.Mailer
+  alias PhoenixTokenAuth.Util
+
+  def user_created_by_username?(changeset) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
+        case Util.repo.insert(changeset) do
+          {:ok, _} -> true
+          {:error, _} -> false
+        end
+      end)
+    end
+  end
+
+  def user_created_and_emailed?(changeset, confirmation_token, conn) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
```

Comment:
here we also don't need the `valid?` check. the transaction should stay, though. we shouldn't insert a record if we can't send the email. otherwise we block the users email without them having a chance of confirming it.